plugins/rbac-backend/src/role-manager/role-list.test.ts

```diff
@@ -0,0 +1,40 @@
+import { RoleList } from './role-list';
+
+describe('RoleList', () => {
+  const roleList = new RoleList('user:default/adam');
+  const newRole = new RoleList('role:default/test');
+  const extraRole = new RoleList('role:default/extra');
+  roleList.addRole(extraRole);
+
+  describe('addRole', () => {
+    it('should add a role', () => {
+      roleList.addRole(newRole);
+
+      expect(roleList.hasRole('role:default/test', 1)).toEqual(true);
+      expect(roleList.hasRole('role:default/does-not-exist', 1)).toEqual(false);
+    });
+
+    it('should not add a duplicate of an existing role', () => {
+      roleList.addRole(newRole);
```

Comment:
This test is the same with 'addRole'

Reply:
Yes, it is the same as the first test but it is only because I am trying to test this [scenario](https://github.com/janus-idp/backstage-plugins/blob/main/plugins/rbac-backend/src/role-manager/role-list.ts#L12:L14) to get 100% test coverage. Since I am not recreating the Role Manager in this test, I am essentially calling `addRole` a second time with the `newRole` to invoke that it there isn't another one added. To further expand on this test, I could potentially check the length of `roleList` to ensure that there are still only two roles. What do you think?